src/test/kotlin/org/jmailen/gradle/kotlinter/functional/AndroidProjectTest.kt

```diff
@@ -0,0 +1,101 @@
+import org.jmailen.gradle.kotlinter.functional.utils.androidManifest
+import org.jmailen.gradle.kotlinter.functional.utils.kotlinClass
+import org.jmailen.gradle.kotlinter.functional.utils.resolve
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import java.io.File
+
+internal class AndroidProjectTest : WithGradleTest.Android() {
+
+    private lateinit var androidModuleRoot: File
+
+    @Before
+    fun setUp() {
+        testProjectDir.root.apply {
+            resolve("settings.gradle") { writeText(settingsFile) }
+            resolve("build.gradle") {
+                // language=groovy
+                val buildScript =
+                    """
+                subprojects {
+                    repositories {
+                        google()
+                        mavenCentral()
+                    }
+                }
+                
+                    """.trimIndent()
+                writeText(buildScript)
+            }
+            androidModuleRoot = resolve("androidproject") {
+                resolve("build.gradle") {
+                    // language=groovy
+                    val androidBuildScript =
+                        """
+                        plugins {
+                            id 'com.android.library'
+                            id 'kotlin-android'
+                            id 'org.jmailen.kotlinter'
+                        }
+                        
+                        android {
+                            compileSdkVersion 31
+                            defaultConfig {
+                                minSdkVersion 23
+                            }
+                            
+                            flavorDimensions 'customFlavor'
+                            productFlavors {
+                                flavorOne {
+                                    dimension 'customFlavor'
+                                }
+                                flavorTwo {
+                                    dimension 'customFlavor'
+                                }
+                            }
+                        }
+                        
+                        """.trimIndent()
+                    writeText(androidBuildScript)
+                }
+                resolve("src/main/AndroidManifest.xml") {
+                    writeText(androidManifest)
+                }
+                resolve("src/main/kotlin/MainSourceSet.kt") {
+                    writeText(kotlinClass("MainSourceSet"))
+                }
+                resolve("src/debug/kotlin/DebugSourceSet.kt") {
+                    writeText(kotlinClass("DebugSourceSet"))
+                }
+                resolve("src/test/kotlin/TestSourceSet.kt") {
+                    writeText(kotlinClass("TestSourceSet"))
+                }
+                resolve("src/flavorOne/kotlin/FlavorSourceSet.kt") {
+                    writeText(kotlinClass("FlavorSourceSet"))
+                }
```

Comment:
If needed, one can also add test case fo `src/xxx/java`, to verify `/java` directory is also recognised properly